server/src/main/java/org/opensearch/index/engine/IngestionEngine.java

```diff
@@ -145,6 +146,16 @@ protected Set<IngestionShardPointer> fetchPersistedOffsets(DirectoryReader direc
 
     @Override
     public IndexResult index(Index index) throws IOException {
+        throw new UnsupportedEncodingException("push-based indexing is not supported in ingestion engine, use streaming source instead");
```

Comment:
Did you intend UnsupportedOperationException instead?

Reply:
that's right